wpilibj/src/main/java/edu/wpi/first/wpilibj/simulation/DCMotorSim.java

```diff
@@ -140,4 +171,30 @@ public double getCurrentDrawAmps() {
   public void setInputVoltage(double volts) {
     setInput(volts);
   }
+
+  /**
+   * Identifies a position and velocity system, based on velocity and acceleration gains.
+   *
+   * <p>This is useful for identifying a state space system for use with {@link DCMotorSim}.
+   *
+   * @param kV The velocity gain, in volts/(unit/sec)
+   * @param kA The acceleration gain, in volts/(unit/sec²)
+   * @return A LinearSystem representing the given characterized constants.
+   * @throws IllegalArgumentException if kV &lt;= 0 or kA &lt;= 0.
+   * @see <a href="https://github.com/wpilibsuite/sysid">https://github.com/wpilibsuite/sysid</a>
+   */
+  public static LinearSystem<N2, N1, N2> identifySystem(double kV, double kA) {
```

Comment:
This is an implementation detail that should be private.

Reply:
Thoughts on a (I would imagine protected) method in LinearSystemId to generate a generic 1dof [position, velocity] system with varying output dimension?

Reply:
`protected` is for exposing functions to derived classes. It's very rarely used.

The problem with adding another overload is that Java erases generic types, so it treats `LinearSystem<N2, N1, N1>` and `LinearSystem<N2, N1, N2>` as the same type for overload resolution. You'd have to give the functions different names.

Reply:
What about an optional argument that takes an ordered list of state indices to put in the measurement model? My main reservation about it is it being error-prone compared to a union type. What we basically want is an ordered set of enum values.

Reply:
I just realized that the models for elevator and single jointed arm sims were completely redundant with `LinearSystemId.identifyPositionSystem`

Reply:
Regarding taking indices for what states to keep, something like this could work if we take an output value in the method signature (of identifyPositionSystem, or a similar method) and build C and D based on that.
D is simple because it will always be a zero vector of length output, but C is more complicated since it is the identity, or part of it.

Reply:
The issue is the size of C must be known at compile time, not runtime.